theano/tensor/tests/test_basic.py

```diff
@@ -7561,6 +7561,88 @@ def test_diag_grad(self):
         tensor.verify_grad(diag, [x], rng=rng)
 
 
+class test_alloc_diag(unittest.TestCase):
+    def __init__(self, name, alloc_diag=AllocDiag, mode=None):
+        self.alloc_diag = AllocDiag
```

Comment:
These tests are always running the cpu version because here you should set it like this:

```python
self.alloc_diag = alloc_diag

```

Reply:
Please rename the class to capitals: `TestAllocDiag`